### PR TITLE
create shop item order

### PR DESCRIPTION
**Pull request by saadao**

### CODE REVIEW

File: src/main/java/AcceptanceTests/ProjectTests/ShopTest.java [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    ShopController mockShopController;
+
+    // @Mock
+    // NotificationService mockNotificationService;
+
+    @BeforeEach
+    void init() {
+        MockitoAnnotations.openMocks(this);
+        // mockLibrary = Mockito.spy(new Library(mockDatabaseService, mockReviewService));
+
```
Comment (inbarbenchaim) [inline src/main/java/AcceptanceTests/ProjectTests/ShopTest.java:38]: remove space

File: src/main/java/AcceptanceTests/ProjectTests/ShopTest.java [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+import static org.mockito.Mockito.*;
+import org.mockito.*;
+
+import ServiceLayer.UserService;
+import ServiceLayer.ShopService;
+import Domain.ShopController;
+import Domain.User;
+
+
+public class ShopTest {
```
Comment (inbarbenchaim) [inline src/main/java/AcceptanceTests/ProjectTests/ShopTest.java:17]: add to the correct folder

File: src/main/java/Domain/ShopController.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
+package Domain;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class ShopController {
+    private static ShopController _shopController;
```
Comment (inbarbenchaim) [inline src/main/java/Domain/ShopController.java:7]: great names!

File: src/main/java/Domain/ShopController.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    private List<Shop> _shopsList;
+
+    private ShopController() {
+        this._shopsList = new ArrayList<>();
+    }
+
+    // Public method to provide access to the _shopController
```
Comment (inbarbenchaim) [inline src/main/java/Domain/ShopController.java:14]: maybe singletone? for later

Comment (inbarbenchaim) [reply src/main/java/Domain/ShopController.java:14]: not need to change now, just think about that

File: src/main/java/Domain/ShopController.java [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+            if (shop.getShopId().equals(shopId)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    public void OpenNewShop(Integer shopId, String userName) throws Exception 
+    {
+        //TODO: need to implement  (check if shop exist)
```
Comment (inbarbenchaim) [inline src/main/java/Domain/ShopController.java:43]: remove todo if done implemented

Comment (inbarbenchaim) [reply src/main/java/Domain/ShopController.java:43]: add todo about permisions

File: src/main/java/ServiceLayer/ShopService.java [hunk 6]
```diff
@@ -1,4 +1,58 @@
...
+        catch (Exception e)
+        {
+            response.setErrorMessage(String.format("Failed to create shopID %d by user %s. Error: ", shopId, userName, e.getMessage()));
+            logger.log(Level.SEVERE, e.getMessage(), e);
+        }
+
+        return response;        
+    }
+
+    public Response addProductToShop(Integer shopId, String userName, Product product)
```
Comment (inbarbenchaim) [inline src/main/java/ServiceLayer/ShopService.java:39]: pls document- its a long function

File: src/main/java/ServiceLayer/ShopService.java [hunk 7]
```diff
@@ -1,4 +1,58 @@
...
+    {
+        Response response = new Response();
+        try
+        {
+            // TODO: verify if register and logged in and verify permissions in Shop
```
Comment (inbarbenchaim) [inline src/main/java/ServiceLayer/ShopService.java:44]: great!

File: src/main/java/ServiceLayer/ShopService.java [hunk 8]
```diff
@@ -1,4 +1,58 @@
...
+            _shopController.addProductToShop(shopId, product, userName);
+            logger.info(String.format("New product %s :: %d added by: %s to Shop ID: %d" ,product.getProductName(), product.getProductId(), userName, shopId));
+
+        }
+        catch (Exception e)
+        {
+            response.setErrorMessage(String.format("Failed to add product %s :: %d to shopID %d by user %s. Error: ", product.getProductName(),
```
Comment (inbarbenchaim) [inline src/main/java/ServiceLayer/ShopService.java:51]: what if the exception is from one of the "gets"? maybe print the args of the function

File: src/main/java/ServiceLayer/ShopService.java [hunk 9]
```diff
@@ -1,4 +1,58 @@
...
+            // TODO: verify if register and logged in and verify permissions in Shop
+            _shopController.addProductToShop(shopId, product, userName);
+            logger.info(String.format("New product %s :: %d added by: %s to Shop ID: %d" ,product.getProductName(), product.getProductId(), userName, shopId));
+
+        }
+        catch (Exception e)
+        {
+            response.setErrorMessage(String.format("Failed to add product %s :: %d to shopID %d by user %s. Error: ", product.getProductName(), 
+            product.getProductId(), shopId, userName, e.getMessage()));
+            logger.log(Level.SEVERE, e.getMessage(), e);
```
Comment (inbarbenchaim) [inline src/main/java/ServiceLayer/ShopService.java:53]: logger.error i think its the same

File: src/main/java/Domain/ShopController.java (hunk 5 above)
Comment (saadao) [reply src/main/java/Domain/ShopController.java:43]: I think that each user the logged in as permission to open new shop

Comment (inbarbenchaim) [reply src/main/java/Domain/ShopController.java:43]: thats true thanks! missed that

**Review by inbarbenchaim — APPROVED** [review]

**Pull request merged**